### PR TITLE
Move induction-like tactics to their own module.

### PR DESCRIPTION
**Pull request by ppedrot**
It's written on the can.

Overlays:
- https://github.com/QuickChick/QuickChick/pull/335

### CODE REVIEW

File: tactics/induction.mli [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Names
+(* open Constr *)
```
Comment (Alizter) [inline tactics/induction.mli:12]: Clean these up?

Comment (ppedrot) [reply tactics/induction.mli:12]: Ah, yes, forgot to do this. I proofread the ml files but not the mli ones...

File: tactics/induction.ml [hunk 2]
```diff
@@ -0,0 +1,1571 @@
...
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+module CVars = Vars
+
+open Pp
```
Comment (Alizter) [inline tactics/induction.ml:13]: While we are here, is it possible to make this list of opens shorter?

Comment (ppedrot) [reply tactics/induction.ml:13]: Even if we did that, I'd rather have it in another PR so as to keep as many lines recognized to have been moved by git as possible.

Comment (Alizter) [reply tactics/induction.ml:13]: Sure thing.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**